tensorflow/core/platform/hadoop/hadoop_file_system.cc

```diff
@@ -191,13 +191,16 @@ Status HadoopFileSystem::Connect(StringPiece fname, hdfsFS* fs) {
                                       nn.empty() ? "default" : nn.c_str());
     cacheKey += nn;
   }
+  mtx_.lock();
```

Comment:
I think a `std::lock_guard` should be a better idea. it will be something like:
```cpp
{
  std::lock_guard l(mutex);
  if (connectionCache_.find(cacheKey) == connectionCache_.end()) {
     connectionCache_[cacheKey] = libhdfs()->hdfsBuilderConnect(builder);
     hdfsFS cacheFs = libhdfs()->hdfsBuilderConnect(builder);
     if (cacheFs == nullptr) {
       return errors::NotFound(strerror(errno));
     }
     connectionCache_[cacheKey] = cacheFs;
   }
   *fs = connectionCache_[cacheKey];
// mutex will be unlocked automatically when the code reachs here
}
```
But we have a tensorflow version mutex and I think we can use it here. https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/platform/mutex.h